### ISSUE TITLE
Helm 3: OCI dependencies not found?

### ISSUE THREAD

**Issue by DanielMe** [helm/helm#9496]
Output of `helm version`:
```
version.BuildInfo{Version:"v3.5.3", GitCommit:"041ce5a2c17a58be0fcd5f5e16fb3e7e95fea622", GitTreeState:"dirty", GoVersion:"go1.15.8"}
```
Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.4", GitCommit:"e87da0bd6e03ec3fea7933c4b5263d151aafd07c", GitTreeState:"clean", BuildDate:"2021-02-18T16:12:00Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.9-eks-d1db3c", GitCommit:"d1db3c46e55f95d6a7d3e5578689371318f95ff9", GitTreeState:"clean", BuildDate:"2020-10-20T22:18:07Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```
Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
AWS. EKS for the cluster (probably not relevant) and ECR for the registry

I'm trying out the experimental OCI support and I stumbled upon something that looks like a bug to me (though I don't want to rule out that I might be using it wrong):

I have pushed a custom chart called `my-chart` to an amazon ECR registry as described [here](https://docs.aws.amazon.com/AmazonECR/latest/userguide/push-oci-artifact.html)

Then, in an "aggregation chart", I'm depending on this artifact using the following dependencies:

```
dependencies:
        - name: my-chart
          version: 0.1.0
          repository: oci://<awsaccount>.dkr.ecr.<region>.amazonaws.com
        - name: kube-prometheus-stack
          version: 14.0.1
          repository: https://prometheus-community.github.io/helm-charts
```

(The prometheus stack is there to have the comparison with a traditional repository)

When I then run `helm dependency update`, both charts are downloaded to the `charts/` directory as expected:

```
$ ls charts/
my-chart-0.1.0.tgz  kube-prometheus-stack-14.0.1.tgz
```

When I then use `helm install` passing each of these charts individually, they are installed on kubernetes as expected (so the charts themselves seem to be okay).
However, when I run `helm install . --generate-name` in the "aggregation chart", I get the following output
```
install.go:173: [debug] Original chart version: ""
install.go:190: [debug] CHART PATH: <...>

Error: found in Chart.yaml, but missing in charts/ directory: my-chart
helm.go:81: [debug] found in Chart.yaml, but missing in charts/ directory: my-chart
helm.sh/helm/v3/pkg/action.CheckDependencies
	/home/circleci/helm.sh/helm/pkg/action/install.go:612
main.runInstall
	/home/circleci/helm.sh/helm/cmd/helm/install.go:216
main.newInstallCmd.func2
	/home/circleci/helm.sh/helm/cmd/helm/install.go:120
github.com/spf13/cobra.(*Command).execute
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850
github.com/spf13/cobra.(*Command).ExecuteC
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958
github.com/spf13/cobra.(*Command).Execute
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
main.main
	/home/circleci/helm.sh/helm/cmd/helm/helm.go:80
runtime.main
	/usr/local/go/src/runtime/proc.go:204
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1374
```

Maybe helm is expecting a different filename convention in case of a chart from an OCI registry?
If I comment out the `my-chart` block and only keep prometheus, the same install command works as expected.

**Comment by jdolitsky** [comment]
Thanks for reporting. Can you try to unpack the mychart tgz file into a "mychart" directory, and let us know if you get the same result?

**Comment by DanielMe** [comment]
Hi @jdolitsky, unpacking the chart into a "mychart" directory leads to the same result. 
However, doing this I observed something interesting: When I subsequently delete all the contents of the "mychart" directory, the result was a different one: Now it complained about there not being a `Chart.yaml`. I concluded that the problem must be in my chart and not in the discovery and sure enough: The name I provided in the `Chart.yaml` did not actually match the name in the registry.
Of course that can't work - problem between keyboard and chair. :slightly_smiling_face: 

So the only possible improvement for helm would be to detect that there is a mismatch between the two and return a better error message but the behavior now appears correct to me. Feel free to close this.
